sunpy/visualization/animator/image.py

```diff
@@ -220,28 +221,67 @@ def __init__(self, data, wcs=None, image_axes=[-1, -2], unit_x_axis=None, unit_y
     def _get_main_axes(self):
         axes = self.fig.add_axes([0.1, 0.1, 0.8, 0.8], projection=self.wcs,
                                  slices=self.slices_wcsaxes)
-        self._set_unit_in_axis(axes)
         return axes
 
-    def _set_unit_in_axis(self, axes):
-        if self.unit_x_axis is not None:
-            axes.coords[2].set_format_unit(self.unit_x_axis)
-            axes.coords[2].set_ticks(exclude_overlapping=True)
-        if self.unit_y_axis is not None:
-            axes.coords[1].set_format_unit(self.unit_y_axis)
-            axes.coords[1].set_ticks(exclude_overlapping=True)
-
     def plot_start_image(self, ax):
         """Sets up plot of initial image."""
         imshow_args = {'interpolation': 'nearest',
                        'origin': 'lower',
                        }
         imshow_args.update(self.imshow_kwargs)
+        ax.ticklabel_format(style='sci')
         im = ax.imshow(self.data[self.frame_index], **imshow_args)
         if self.if_colorbar:
             self._add_colorbar(im)
         return im
 
+    def update_axes_style(self, tick_spacing, x_tick_position=None,
+                          y_tick_position=None,  xlabel=None, ylabel=None):
+        """
+        Allows a user to change the axes' units, labels and ticks of an existing visualization.
+
+        Parameters
+        ----------
+
+        tick_spacing: `astropy.units`
+            Spacing between two consecutive ticks
+        x_tick_position: `str`
+            Tick position for the X axis
+        y_tick_position: `str`
+            Tick position for the Y axis
+        xlabel: `str`
+            Label for X axis in the plot
+        ylabel: `str`
+            Label for Y axis in the plot
+
+        """
+        ax = self.axes
+        if self.unit_x_axis is not None:
+            ax.coords[2].set_format_unit(self.unit_x_axis)
+        if self.unit_y_axis is not None:
+            ax.coords[1].set_format_unit(self.unit_y_axis)
+        if not isinstance(tick_spacing, u.Quantity):
+            raise TypeError("Astropy Quantities should be provided")
+        else:
+            ax.coords[2].set_ticks(spacing=tick_spacing, exclude_overlapping=True)
+            ax.coords[1].set_ticks(spacing=tick_spacing, exclude_overlapping=True)
+        if xlabel is None:
+            ax.coords[2].set_axislabel(self.wcs.wcs.ctype[0])
+        else:
+            ax.coords[2].set_axislabel(xlabel)
+        if ylabel is None:
+            ax.coords[1].set_axislabel(self.wcs.wcs.ctype[1])
+        else:
+            ax.coords[1].set_axislabel(ylabel)
```

Comment:
`These two `if` blocks can be _dry_-er with a loop and `enumerate`, for example:
```python
for i,axis_label in enumerate([ylabel, xlabel], start=1):
    if axis_label is None:
        ax.coords[i].set_axislabel(self.wcs.wcs.ctype[i%2])
    else:
        ax.coords[i].set_axislabel(axislabel)
```